docs/specs/oidc.md

```diff
@@ -489,4 +520,38 @@ When the app session token is consumed:
 - The session cookie would contain a token referencing the refresh token,
   instead of IdP sessions. Therefore, the lifetime of session cookie is bound
   to refresh token instead of IdP session.
- 
\ No newline at end of file
+
+## How to construct authentication request to achieve different scenarios
+
+### The user has NOT signed in yet in my mobile app. I want to authenticate any user.
+
+The authentication request is `response_type=code&prompt=login`.
+
+The user will NOT see select account screen even if they have previously signed in.
+
+The user is authenticated fully.
+
+### The user has NOT signed in yet in my mobile app. I want to authenticate any user. Possibly reuse any previous signed in sessions.
+
+The authentication request is `response_type=code`.
+
+The user will see select account screen if they have valid sessions.
+If the user continues with an existing valid session, the user is NOT reauthenticated.
+If the user continues with another account, the user is authenticated fully.
+
+### The user has signed in. I want to reauthenticate the user before they can perform sensitive operation.
+
+The authentication request is `response_type=id_token&prompt=login&max_age=0&id_token_hint=ID_TOKEN&acr_values=https://authgear.com/acr?reauthenticate=primary`.
+
+The user will NOT see select account screen because `id_token_hint` is given.
+
+If the user has valid session, the user is reauthenticated with one of their primary authenticator.
+It is an error if the user does not have any primary authenticator.
```

Comment:
Could we have some examples for this case? e.g. reauthenticate a user who only connected to OAuth?